Commit CRTMv3 configuration (backward compatible with 2.4)

karpob opened the pull request:
## Description

Adds CRTMv3 capability while maintaining 2.4 compatibility. Also modernizes setup removing deprecated (or soon to be deprecated) "python setup.py install" method in favor of "pip install . " method.  Adds helper scripts to install CRTMv3 + pyCRTM, along with dependencies using miniconda.  

Also resolves #17 fixing issues related to setuptools version changes, along with f2py issues. This is done by replacing f2py executable which has bug related to passing preprocessor directives with "python -m numpy.f2py"  which passes preprocessor directives correctly. 

## Issue(s) addressed

Resolves #17

## Dependencies

List the other PRs that this PR is dependent on:
- none

## Impact

Expected impact on downstream repositories:
None

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

BenjaminTJohnson commented on setup.py:
```diff
@@ -9,137 +8,80 @@ def main():
+    for root,dirs,filez in os.walk(searchPath):
+        for name in filez:
+            srcPath = os.path.join(root,name)
+            if(not ('ODAS' in srcPath or 'Big_Endian' in srcPath)):
```
@karpob as an FYI: future CRTMv3 after v3.2.x will be netCDF only.

karpob replied:
Yeah my thought is when it's netcdf only to move all of that out of setup.py into something like cartopy does for higher resolution maps where it downloads on request when it doesn't find anything. Also, at some point figure out how to remove setup.py completely.

chengdang commented:
Hello Bryan, This is cool! I like make_it_so.sh very much. I used `./make_it_so.sh apple_silicon ` with M2 chip

There are a few hiccups though that I had to manually install some libraries, which seems to be caused by the failure of activate a virtual environment with the following command.

```
export CONDA_VENV='pycrtm'
conda create --name ${CONDA_VENV} python scikit-build h5py netcdf4 gfortran libnetcdf netcdf-fortran meson cmake git git-lfs matplotlib
Channels:w
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - netcdf-fortran

Current channels:

  - defaults
  - https://repo.anaconda.com/pkgs/main
  - https://repo.anaconda.com/pkgs/r

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

I'm able to create a virtual environment manually without lib appendix and install netcdf-fortran with conda forge though:

`conda install conda-forge::netcdf-fortran
`

All other libs works with conda create.

karpob commented:
@chengdang Good catch! I forgot to move my .condarc when I was testing this (I usually just put conda-forge by default by habit to avoid licensing issues.) I added a line `conda config --add channels conda-forge` that seems to take care of it when I ran a test (duplicating your issue, then fixing it with the extra line).

chengdang reviewed (approved):
Thanks for the quick fix! Everything works as expected now. I am able to create pycrtm, and run the example. Thank you!

karpob commented:
Also added fix per #20 where LD_LIBRARY_PATH wasn't set automatically as designed, when using shared libraries.

karpob commented:
I investigated some of the issues @madison-BEST was experiencing with miniconda under linux. I think I solved those by forcing -fPIC.  While doing it I went back and double checked backward compatibility and found it was broken. I fixed it. To preserve my sanity I added a hidden option to install build against 2.4.1 in `make_it_so.sh.` Sorry for the multiple revisions on this pull request!

chengdang commented:
With this branch, I encountered consistent Kernel crash for visible sensors, for example, if set `sensor_id = 'v.abi_g18'`in example `test_atms.ipynb`, Kernel crashed after executing `crtmOb.loadInst()`:

```
crtmOb.loadInst()

The Kernel crashed while executing code in the current cell or a previous cell. 
Please review the code in the cell(s) to identify a possible cause of the failure. 
Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. 
View Jupyter [log](command:jupyter.viewOutput) for further details.
```


I've added a few print statements in function `loadInst`, the function itself is running fine till the last line, then the Kernel crashed. I tried both binary and nc files, Kernal crashed regardless of file format. So far I only seen this with visible sensors. Not sure why.

karpob commented:
I do. Well, I think I do, as duplicate the error, but it dies after the call to run_direct.  The coefficient linking only grabs ODPS, not ODAS, so the SpcCoeffs are there, but not the TauCoeffs. If you do the equivalent in `test_atms.py` you'll get:
Inquire_AlgorithmID_nc4(FAILURE) : File /Users/bkarpowi/github/crtm_20250505/pycrtm/ext/crtm_coefficients/v.abi_g18.TauCoeff.nc not found.
 CRTM_Load_TauCoeff(FAILURE) : cannot obtain transmittance algorithm ID from file /Users/bkarpowi/github/crtm_20250505/pycrtm/ext/crtm_coefficients/v.abi_g18.TauCoeff.nc
 CRTM_Init(FAILURE) : Error loading TauCoeff data
 Error Initializing CRTM

I can add a check to the linking to search each SpcCoeff to find a TauCoeff pair, if it's missing.

karpob commented:
![image](https://github.com/user-attachments/assets/1a5ef5b4-a916-4b71-b533-4134ee685cbc)

karpob commented:
The last push should fix the missing visible Tau files.
